src/Adapter/Product/Image/Repository/ProductImageRepository.php

```diff
@@ -94,28 +87,25 @@ public function __construct(
         string $dbPrefix,
         ProductImageValidator $productImageValidator,
         ProductImagePathFactory $productImagePathFactory,
-        CombinationRepository $combinationRepository,
-        ProductMultiShopRepository $productMultiShopRepository
+        CombinationRepository $combinationRepository
     ) {
         $this->connection = $connection;
         $this->dbPrefix = $dbPrefix;
         $this->productImageValidator = $productImageValidator;
         $this->productImagePathFactory = $productImagePathFactory;
         $this->combinationRepository = $combinationRepository;
-        $this->productMultiShopRepository = $productMultiShopRepository;
     }
 
     /**
      * @param ProductId $productId
-     * @param ShopConstraint $shopConstraint
+     * @param array $shopIds
      *
      * @return Image
      *
      * @throws CoreException
      * @throws ProductImageException
-     * @throws CannotAddProductImageException
      */
-    public function create(ProductId $productId, ShopConstraint $shopConstraint): Image
+    public function create(ProductId $productId, array $shopIds): Image
```

Comment:
There is also a `create` method in `ProductImageMultiShopRepository`, it seems like only one should be used now so the other should be removed if it's not used anymore (this should be the one in theory)